core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/file/UserFileResource.scala

```diff
@@ -258,4 +258,34 @@ class UserFileResource {
 
   }
 
+  /**
+    * This method updates the name of a given userFile
+    *
+    * @param session HttpSession
+    * @return the updated userFile
+    */
+  @POST
+  @Path("/update/name")
+  @Consumes(Array(MediaType.APPLICATION_JSON))
+  @Produces(Array(MediaType.APPLICATION_JSON))
+  def changeUserFileName(file: File, @Auth sessionUser: SessionUser): Response = {
+    val user = sessionUser.getUser
+    val fid = file.getFid
+    val newFileName = file.getName
+
+    val validationRes = this.validateFileName(newFileName, user.getUid)
+    if (validationRes.getLeft) {
+      val userFile = fileDao.fetchOneByFid(fid)
+      userFile.setName(newFileName)
+      fileDao.update(userFile)
+      Response.ok(userFile).build()
```

Comment:
This is a very high level comment: I haven't checked other details yet. But does changing this file name also changes the file saved on the disk? I think you did not apply the change to the file system. Thus I assume the file with the new name cannot be executed?